openmapflow/labeled_dataset.py

```diff
@@ -79,53 +320,204 @@ class LabeledDataset:
     to train and evaluate a macine learning model a model.
-
-    Args:
-        dataset (str): The name of the dataset.
-        country (str): The country of the dataset (can be 'global' for global datasets).
     """
 
-    dataset: str = ""
-    country: str = ""
-    label_type: str = "binary"
-    license: str = ""
-    source: str = ""
-    dataset_dir: Path = PROJECT_ROOT / dp.DATASETS
-
     def __post_init__(self):
-        self.df_path = self.dataset_dir / (self.dataset + ".csv")
+        np.random.seed(SEED)
+        random.seed(SEED)
+        self.name = self.__class__.__name__
+        if self.name == "LabeledDataset":
+            raise ValueError("LabeledDataset must be inherited to be used.")
+        self.df_path = PROJECT_ROOT / dp.DATASETS / (self.name + ".csv")
+
+    def load_labels(self) -> pd.DataFrame:
+        raise NotImplementedError
 
     def summary(self, df: pd.DataFrame) -> str:
         timesteps = get_label_timesteps(df).unique()
         eo_status_str = str(df[EO_STATUS].value_counts()).rsplit("\n", 1)[0]
         return (
-            f"{self.dataset} (Timesteps: {','.join([str(int(t)) for t in timesteps])})\n"
+            f"{self.name} (Timesteps: {','.join([str(int(t)) for t in timesteps])})\n"
             + "----------------------------------------------------------------------------\n"
             + eo_status_str
             + "\n"
             + _label_eo_counts(df)
             + "\n"
         )
 
-    def load_df(
-        self, skip_to_np: bool = False, check_eo_data: bool = True
-    ) -> pd.DataFrame:
+    def _duplicates_check(self, df: pd.DataFrame):
+        clean_df = df[clean_df_condition(df)]
+        duplicates = clean_df.duplicated(subset=[EO_LAT, EO_LON, EO_FILE])
+        if duplicates.sum() > 0:
+            print(f"Found {duplicates.sum()} duplicates")
+            duplicates_index = clean_df.index[duplicates]
+            df.loc[duplicates_index, EO_STATUS] = EO_STATUS_DUPLICATE
+            df.loc[duplicates_index, EO_LAT] = None
+            df.loc[duplicates_index, EO_LON] = None
+            df.loc[duplicates_index, EO_FILE] = None
+            df.to_csv(self.df_path, index=False)
+        return df
+
+    def load_df(self, check_eo_data: bool = True, to_np: bool = False) -> pd.DataFrame:
         """Load dataset (labels + earth observation data) as a DataFrame"""
         if not self.df_path.exists():
             print(self.create_dataset())
         df = pd.read_csv(self.df_path)
         df = df[clean_df_condition(df)].copy()
         if check_eo_data and df[EO_DATA].isnull().any():
             raise ValueError(
-                f"{self.dataset} has missing earth observation data, "
+                f"{self.name} has missing earth observation data, "
                 + "run openmapflow create-datasets"
             )
-        if not skip_to_np:
-            df[EO_DATA] = df[EO_DATA].apply(_to_np)
+        if to_np:
```

Comment:
Changed default to not convert to_np

Reply:
why? Was it taking too long?

Seems more useful to have usable np arrays by default.

Reply:
These are the primary dataframe use cases I can think of:
1. Exploratory data analysis (one or a couple of rows at a time)
2. Training ( >80% of the rows at a time)
3. Evaluation (<20% of the rows at a time)

Loading all Geowiki points into memory takes 25s and it's only necessary for 2. 
It's easier to just load what's needed for 1 and 3, hence the default

<img width="948" alt="image" src="https://user-images.githubusercontent.com/25067660/191133586-f2636ca9-868b-45d6-be8a-319ef64a4ca6.png">